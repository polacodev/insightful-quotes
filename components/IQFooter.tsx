import React from "react";
import { View, Text } from "react-native";

const IQFooter = () => {
  return (
    <View className="justify-center items-center">
      <Text style={{ fontFamily: 'NunitoSans-Italic', fontSize: 10 }} className="text-xs text-center text-[#b0b0b0]">This application uses the FavQs API.</Text>
      <Text style={{ fontFamily: 'NunitoSans-Italic', fontSize: 10 }} className="text-xs text-center text-[#b0b0b0]">All rights belong to their respective owners</Text>
      <Text style={{ fontFamily: 'NunitoSans-Italic', fontSize: 8 }} className="text-xs text-[#b0b0b0]">@polacodev</Text>
    </View>
  );
};

export default IQFooter;
