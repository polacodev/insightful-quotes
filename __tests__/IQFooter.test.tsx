import { render } from "@testing-library/react-native";

import IQFooter from "@/components/IQFooter";

describe("<IQFooter />", () => {
  test("IQFooter renders correctly the texts and Snapshot", () => {
    const { getByText, toJSON } = render(<IQFooter />);
    const text1 = getByText("This application uses the FavQs API.")
    const text2 = getByText("All rights belong to their respective owners")
    const text3 = getByText("@polacodev")

    expect(text1).toBeTruthy();
    expect(text2).toBeTruthy();
    expect(text3).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
