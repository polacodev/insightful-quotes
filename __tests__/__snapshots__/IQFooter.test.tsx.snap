// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IQFooter /> IQFooter renders correctly the texts and Snapshot 1`] = `
<View
  className="justify-center items-center"
>
  <Text
    className="text-xs text-center text-[#b0b0b0] dark:text-[#676767]"
    style={
      {
        "fontFamily": "NunitoSans-Italic",
        "fontSize": 10,
      }
    }
  >
    This application uses the FavQs API.
  </Text>
  <Text
    className="text-xs text-center text-[#b0b0b0] dark:text-[#676767]"
    style={
      {
        "fontFamily": "NunitoSans-Italic",
        "fontSize": 10,
      }
    }
  >
    All rights belong to their respective owners
  </Text>
  <Text
    className="text-xs text-[#b0b0b0] dark:text-[#676767]"
    style={
      {
        "fontFamily": "NunitoSans-Italic",
        "fontSize": 8,
      }
    }
  >
    @polacodev
  </Text>
</View>
`;
